compiler/noirc_evaluator/src/ssa/opt/constant_folding.rs

```diff
@@ -192,8 +192,32 @@ impl Context {
         }
 
         // Resolve any inputs to ensure that we're comparing like-for-like instructions.
+        let instruction = instruction
+            .map_values(|value_id| resolve_cache(dfg, constraint_simplification_mapping, value_id));
+
+        // Sort operands of commutative instructions
+        if let Instruction::Binary(binary) = &instruction {
+            match binary.operator {
```

Comment:
Oh, I meant sorting when the instructions are first created. Not during constant folding.